docs/how-to/c-functions.rst

```diff
@@ -0,0 +1,375 @@
+
+However, unlike Objective-C methods, we cannot call C functions right away --- we must first declare the function's argument and return types. (Rubicon cannot do this automatically like with Objective-C methods, because plain C doesn't provide the runtime type information necessary for this.) This type information is found in C header files, in this case ``stdio.h`` (which defines standard C input/output functions, including ``puts``).
+
+The exact location of the macOS C headers varies depending on your version of macOS and the developer tools --- it is not a fixed path. To open the header directory in the Finder, run the following command in the terminal:
+
+.. code-block:: sh
+
+    $ open "$(xcrun --show-sdk-path)/usr/include"
+
+.. note::
+
+    This command requires a version of the macOS developer tools to be installed. If you do not have Xcode or the command-line developer tools installed yet, run this command in the terminal to install the command-line developer tools:
+
+    .. code-block:: sh
+
+        $ xcode-select --install
+
+Once you have opened the relevant header file in a text editor, you need to search for the declaration of the function you're looking for. In the case of ``puts``, it looks like this:
+
+.. code-block:: c
+
+    int puts(const char *);
+
+This means that ``puts`` returns an ``int`` and takes a single argument of type ``const char *`` (a C string). This translates to the following Python ``ctypes`` code:
+
+.. code-block:: python
+
+    >>> from ctypes import c_char_p, c_int
+    >>> libc.puts.restype = c_int
+    >>> libc.puts.argtypes = [c_char_p]
+
+Now that we have provided all of the necessary type information, we can call ``libc.puts``:
+
+.. code-block:: python
+
+    >>> libc.puts(b"Hello!")
+    Hello!
+    10
+
+.. note::
+
+    The ``10`` is the return value of ``puts``, which is printed out by the Python REPL; you can ignore it. (A non-negative return value from ``puts`` indicates that the call was successful.)
+
+    If you're running this code from an editor or IDE and you see only the ``10`` and not the ``Hello!``, try running the code from a Python REPL in a terminal window instead. Some editors/IDEs, such as Python's IDLE, can only capture and display output produced by high-level Python functions (such as ``print``), but not output from low-level C functions (such as ``puts``).
+
+Most real examples of C functions are more complicated than ``puts``, but the basic procedure for calling them is the same: import or load the function's C library, set the function's return type and argument types based on the relevant header, and then call the function as needed.
+
+Each C library only needs to be imported/loaded once, and the ``restype`` and ``argtypes`` only need to be set once per function. This is usually done at module level near the beginning of the module, similar to Python imports.
+
+Inline functions: ``NSLocationInRange``
+---------------------------------------
```

Comment:
Might be worth rephrasing this as "An inline function example"; as written, this reads as if NSLocationInRange is the only inline function that is available.